aom-sys/build.rs

```diff
@@ -1,20 +1,74 @@
-extern crate bindgen;
-extern crate metadeps;
+// build.rs
 
-#[cfg(feature="build")]
+// extern crate cmake;
+#[cfg(unix)]
+extern crate pkg_config;
+#[cfg(unix)]
+extern crate bindgen;
 extern crate cmake;
 
-use std::fs::OpenOptions;
-use std::io::Write;
+use std::env;
+use std::fs;
+use std::path::Path;
 
-fn format_write(builder: bindgen::Builder, output: &str) {
+fn main() {
+    let cargo_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+    let build_path = Path::new(&cargo_dir).join("data/aom");
+    let out_dir = Path::new(&cargo_dir).join("data/aom_build");
+    let debug = if let Some(v) = env::var("PROFILE").ok() {
+        match v.as_str() {
+            "bench" | "release" => false,
+            _ => true,
+        }
+    } else {
+        false
+    };
+
+    let dst = cmake::Config::new(build_path)
+        .define("CONFIG_DEBUG", (debug as u8).to_string())
+        .define("CONFIG_ANALYZER", "0")
+        .define("ENABLE_DOCS", "0")
+        .define("ENABLE_TESTS", "0")
+        .out_dir(out_dir)
+        .no_build_target(cfg!(windows))
+        .build();
+
+    // Dirty hack to force a rebuild whenever the defaults are changed upstream
+    // let _ = fs::remove_file(dst.join("build/CMakeCache.txt"));
+
+    env::set_var("PKG_CONFIG_PATH", dst.join("lib/pkgconfig"));
+    let _libs = pkg_config::Config::new().statik(true).probe("aom").unwrap();
+
+    use std::io::Write;
+
+    let out_dir = env::var("OUT_DIR").unwrap();
+
+    let headers = _libs.include_paths.clone();
+
+    let mut builder = bindgen::builder()
+        .raw_line("#![allow(dead_code)]")
```

Comment:
The new bindgen best practice requires to do in a [different](https://rust-lang-nursery.github.io/rust-bindgen/tutorial-4.html) way. While at it could you please rework it this way? :)